mypy/build.py

```diff
@@ -229,7 +229,7 @@ def default_data_dir(bin_dir: Optional[str]) -> str:
     """
     if not bin_dir:
         if os.name == 'nt':
-            prefixes = [os.path.join(sys.prefix, 'Lib'), os.path.join(site.getuserbase(), 'lib')]
+            prefixes = [os.path.join(sys.prefix, 'Lib'), os.path.join(get_python_lib(), 'lib')]
```

Comment:
Sadly, this will not work on `--user` installs. It might be better to try`site.getuserbase()`, then fall back on sysconfig's get_python_lib, so it is only invoked in virtualenvs.

Reply:
That is too bad.
Do you think that the following solution could do the trick?
``` python
getattr(site, 'getuserbase', None)
prefixes = [
    os.path.join(sys.prefix, 'Lib'), 
    os.path.join(getattr(site, 'getuserbase', get_python_lib)(), 'lib')
]
```

Reply:
Hm, I think the getattr call, while concise, is a bit opaque and not obvious. I think
```python
prefixes = [os.path.join(sys.prefix, 'Lib')]
try:
    prefixes.append(os.path.join(site.getuserbase(), 'lib')
except:
    prefixes.append(os.path.join(get_python_lib(), 'lib')
``` 
or perhaps

```python
prefixes = [os.path.join(sys.prefix, 'Lib')]
if hasattr(site, 'getuserbase'):
    prefixes.append(os.path.join(site.getuserbase(), 'lib')
else:
    prefixes.append(os.path.join(get_python_lib(), 'lib')
```

would be much more readable.

Reply:
Your suggestion looks way more readable, thanks!
Will implement it *soonTM* (in about 12 hours).

Reply:
Also, if you could, explain _why_ we fall back on `get_python_lib` please. eg in the except/else branch
`# getuserbase not available in virtualenvs`

Thanks so much for fixing this!

Reply:
No problem, glad to help and thanks for your quick feedback!